flask_user/forms.py

```diff
@@ -210,7 +221,7 @@ def validate(self):
             user, user_email = user_manager.find_user_by_email(self.email.data)
 
         # Handle successful authentication
-        if user and user.password and user_manager.verify_password(self.password.data, user):
+        if user and user_manager.verify_password(self.password.data, user):
```

Comment:
@3noch - Your assertions are correct about form data - the problem is that when a user doesn't have a password and then they attempt to log in - an error is raised in the call to user_manager.verify_password.  I'd forgotten the details when replying earlier today, but it's easy to reproduce - clear the db password for a user and attempt to log in - at the bottom of the error stack:

```
  File ".../env/local/lib/python2.7/site-packages/passlib/context.py", line 1447, in identify_record
    raise ExpectedStringError(hash, "hash")
TypeError: hash must be unicode or bytes, not None
```
The point of this check is clear - if the user doesn't have a password, don't bother trying to verify.  Yes, you could alternatively patch the lower level checks, but this seemed like the clean work around to me.